### PR TITLE
Adds the Juniper R kernel.

### PR DESCRIPTION
**Pull request by freuk**
Try it out:

```
nix-shell -E '
let
  jupyterWithSrc = builtins.fetchGit {
    url = https://github.com/freuk/jupyterWith;
    ref = "refs/heads/juniper";
    rev = "";
  };
  jupyter = import jupyterWithSrc {};
  jupyterlabWithKernels =
    jupyter.jupyterlabWith {
      kernels = with jupyter.kernels; [
        ( juniperWith {
            name = "base";
            packages = p: with p; [ ggplot2 ];
          })
      ];
    };
in
  jupyterlabWithKernels.env ' --run "jupyter notebook"

```

Cheers.

### CODE REVIEW

**Comment by MMesch** [comment]
Nice! I just tried it and it looks good. Do you want to add short example and an image for the kernel? Or do you prefer that I do it?

File: kernels/juniper.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  jKernel = stdenv.mkDerivation {
+    name = "juniper";
+    phases = "installPhase";
+    src = ./R.svg;
+    buildInputs = [];
+    installPhase = ''
+      mkdir -p $out/kernels/juniper_${name}
+      cp $src $out/kernels/juniper_${name}/logo-64x64.png
```
Comment (MMesch) [inline kernels/juniper.nix]: `       cp $src $out/kernels/juniper_${name}/logo-64x64.png `

this should be `.svg`

**Comment by freuk** [comment]
Done, thanks :+1:

File: kernels/R.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<html lang="en">  <head> <meta content="origin" name="referrer">  <script nonce="8RGD/Pg9mJ5CByzbnCFKMw==">window.google = {};(function(){var e=function(){this.g="";this.h=c};e.prototype.i=!0;var f=/^(?:(?:https?|mailto|ftp):|[^:/?#]*(?:[/?#]|$))/i,c={},g=function(b){var d=new e;d.g=b;return d};g("about:blank");google.navigateTo=function(b,d,a){b!=d&&b.google?b.google.r&&(b.google.r=0,b=b.location,a instanceof e||a instanceof e||(a="object"==typeof a&&a.i?a.g:String(a),f.test(a)||(a="about:invalid#zClosurez"),a=g(a)),b.href=a instanceof e&&a.constructor===e&&a.h===c?a.g:"type_error:SafeUrl",d.location.replace("about:blank")):d.location.replace(a)};}).call(this);(function(){var redirectUrl='https://commons.wikimedia.org/wiki/File:R_logo.svg';google.navigateTo(parent,window,redirectUrl);})();</script> <noscript> <META content="0;URL=https://commons.wikimedia.org/wiki/File:R_logo.svg" http-equiv="refresh"></META> </noscript> </head>  </html>
```
Comment (MMesch) [inline kernels/R.svg]: something is wrong with the svg. Can we use this logo instead: https://github.com/JuniperKernel/JuniperKernel/blob/master/extras/juniper_logo.png ?

Comment (MMesch) [reply kernels/R.svg]: the current svg seems to be some kind of preview script from google.

**Comment by MMesch** [comment]
just checked it again and the notebook runs very well. The only thing missing is the working logo. Plots don't show up in jupyterlab, because it would need some kind of extension that I can't find. This is another issue however and I think we should merge once we have the logo.

Btw stupid question: what is the difference between the IRkernel and the Juniper kernels?

File: kernels/R.svg (hunk 2 above)
Comment (freuk) [reply kernels/R.svg]: Yeah I was expeditive, there you go.

**Comment by freuk** [comment]
> just checked it again and the notebook runs very well. The only thing missing is the working logo. Also plots don't show up in jupyterlab because juniper would need some kind of extension that I can't find (maybe they don't exist yet). This is a different issue however, unrelated to the kernel implementation, and I think we should merge once we have the logo.
Plots should show, at least those in the example notebook do. No extension should be needed afaik.
> 
> Btw stupid question: what is the difference between the IRkernel and the Juniper kernels?
Mostly none other than being Bloomberg trying to push their OSS presence

**Comment by MMesch** [comment]
are you sure that it works in jupyterlab ? Executing `nix-shell --command "jupyter lab" ? I get exactly this issue: https://github.com/JuniperKernel/JuniperKernel/issues/6 .

If there is already a fix, we might just need to use a newer version of it in nixpkgs. I'll merge this version already because the notebook works fine.

**Comment by freuk** [comment]
Oh yeah, the same happens for me. I didn't notice that, good to know. Thanks :)

**Pull request merged**